test/functional/feature_segwit.py

```diff
@@ -197,21 +195,21 @@ def run_test(self):
         assert_equal(self.nodes[1].getbalance(), 20 * Decimal("49.999"))
         assert_equal(self.nodes[2].getbalance(), 20 * Decimal("49.999"))
 
-        self.generate(self.nodes[0], 260)  # block 423
+        self.generate(self.nodes[0], 264)  # block 427
 
-        self.log.info("Verify witness txs are skipped for mining before the fork")
-        self.skip_mine(self.nodes[2], wit_ids[NODE_2][P2WPKH][0], True)  # block 424
-        self.skip_mine(self.nodes[2], wit_ids[NODE_2][P2WSH][0], True)  # block 425
-        self.skip_mine(self.nodes[2], p2sh_ids[NODE_2][P2WPKH][0], True)  # block 426
-        self.skip_mine(self.nodes[2], p2sh_ids[NODE_2][P2WSH][0], True)  # block 427
+        self.log.info("Verify witness txs cannot be mined before the fork")
+        self.fail_mine(self.nodes[2], wit_ids[NODE_2][P2WPKH][0], True)
+        self.fail_mine(self.nodes[2], wit_ids[NODE_2][P2WSH][0], True)
+        self.fail_mine(self.nodes[2], p2sh_ids[NODE_2][P2WPKH][0], True)
+        self.fail_mine(self.nodes[2], p2sh_ids[NODE_2][P2WSH][0], True)
 
         self.log.info("Verify unsigned p2sh witness txs without a redeem script are invalid")
         self.fail_accept(self.nodes[2], "mandatory-script-verify-flag-failed (Operation not valid with the current stack size)", p2sh_ids[NODE_2][P2WPKH][1], sign=False)
         self.fail_accept(self.nodes[2], "mandatory-script-verify-flag-failed (Operation not valid with the current stack size)", p2sh_ids[NODE_2][P2WSH][1], sign=False)
 
-        self.generate(self.nodes[2], 4)  # blocks 428-431
+        self.generate(self.nodes[0], 4)  # blocks 428-431
```

Comment:
does this test not care which node mines these blocks?

Reply:
No. This is only used to increase the block height, so that the activation height is hit. See also:

```py
                "-testactivationheight=segwit@432",